src/test/java/com/endava/internship/collections/TestStudentList.java

```diff
@@ -0,0 +1,483 @@
+    }
+
+    @Test
+    @DisplayName("Check throwing of IndexOutOfBoundsException if 'to > arraySize' in subListRangeCheck()")
+    public void checkThrowingOf_IndexOutOfBoundsException_IfToArraySizeInSubListRangeCheck() {
+        assertThrows(IndexOutOfBoundsException.class, () -> students_List.subList(students_List.size() - 1, students_List.size() + 1));
+    }
+
+    @Test
+    @DisplayName("Check throwing of IllegalArgumentException if 'from > to' in subListRangeCheck()")
+    public void checkThrowingOf_IllegalArgumentException_IfFromToInSubListRangeCheck() {
+        assertThrows(IllegalArgumentException.class, () -> students_List.subList(students_List.size() + 1, students_List.size()));
+    }
+
+    @Test
+    @DisplayName("Check throwing of IndexOutOfBoundsException if 'from < 0' in subListRangeCheck()")
+    public void checkThrowingOf_IndexOutOfBoundsException_IfFromToInSubListRangeCheck() {
+        assertThrows(IndexOutOfBoundsException.class, () -> students_List.subList(-1, 2));
+    }
+
+    @Test
+    @DisplayName("Verify T[] toArray(final T[] input)")
+    public void verifyGenericToArray() {
+        Student[] students = new Student[students_List.size()];
+        students = students_List.toArray(students);
+
+        assertEquals(students.length, students_List.size());
+        assertEquals(students[0], students_List.get(0)); // Check first objects
+        assertEquals(students[students.length - 1], students_List.get(students_List.size() - 1)); // Check last objects
+    }
+
+    @Test
+    @DisplayName("Verify T[] toArray(final T[] input) when input.length < size()")
+    public void verifyTToArrayFinalTInputWhenInputLengthSize() {
+        Student[] students = new Student[students_List.size() / 2];
+        int sizeBefore = students.length;
+        students = students_List.toArray(students);
+
+        assertNotEquals(sizeBefore, students_List.size()); // Check that size are not same
+        assertEquals(students.length, students_List.size()); // Check that size are same
+        assertEquals(students[0], students_List.get(0)); // Check first objects
+        assertEquals(students[students.length - 1], students_List.get(students_List.size() - 1)); // Check last objects
+    }
+
+    @Test
+    @DisplayName("Add object to the index")
+    public void addObjectToTheIndex() {
+        int index = students_List.size() - 3;
+        students_List.add(index, student0);
+
+        assertEquals(student0, students_List.get(index));
+    }
+
+    @Test
+    @DisplayName("Set object to the specified index")
+    public void setObjectToTheSpecifiedIndex() {
+        int index = students_List.indexOf(students_List.get(students_List.size() / 2));
+        Student studentAtIndex = students_List.get(index);
+
+        students_List.set(index, student0);
+
+        assertFalse(studentAtIndex.equals(students_List.get(index)));
+        assertTrue(student0.equals(students_List.get(index)));
```

Comment:
can be simplified
![image](https://user-images.githubusercontent.com/65505271/134980133-2649a3b8-76d1-479c-861e-20aa2f67933b.png)